packages/ai-jsx/src/core/conversation.tsx

```diff
@@ -160,28 +160,40 @@ export function isConversationalComponent(element: AI.Element<any>): boolean {
 }
 
 function toConversationMessages(partialRendering: AI.PartiallyRendered[]): ConversationMessage[] {
-  return partialRendering
-    .flatMap((e) => (AI.isElement(e) ? [e] : []))
-    .map<ConversationMessage>((e) => {
-      switch (e.tag) {
-        case UserMessage:
-          return { type: 'user', element: e };
-        case AssistantMessage:
-          return { type: 'assistant', element: e };
-        case SystemMessage:
-          return { type: 'system', element: e };
-        case FunctionCall:
-          return { type: 'functionCall', element: e };
-        case FunctionResponse:
-          return { type: 'functionResponse', element: e };
-        default:
-          throw new AIJSXError(
-            `Unexpected tag (${e.tag.name}) in conversation`,
-            ErrorCode.UnexpectedPartialRenderResult,
-            'internal'
-          );
+  const invalidChildren = partialRendering.filter((el) => typeof el === 'string' && el.trim()) as string[];
```

Comment:
@petersalas I moved this check to `toConversationMessages` rather than `openai` – does that seem right?

Reply:
I think so, yeah